File: LeetCode/Medium/0852-peak-index-in-a-mountain-array/0852-peak-index-in-a-mountain-array.c
int peakIndexInMountainArray(int* arr, int arrSize){
    int find_max(int start, int end){
        if(start < end){
            int mid = (start + end) / 2;
            if(arr[mid] < arr[mid+1])
                return find_max(mid + 1, end);
            else
                return find_max(start, mid);
        }
        return start;
    }
    return find_max(0, arrSize-1);
}

File: LeetCode/Medium/0852-peak-index-in-a-mountain-array/0852-peak-index-in-a-mountain-array.py
class Solution:
    def peakIndexInMountainArray(self, arr: List[int]) -> int:
        def find_max(start, end):
            mid = (start + end)//2
            if start < end:
                if arr[mid] > arr[mid+1]:
                    return find_max(start, mid)
                else:
                    return find_max(mid+1, end)
            else:
                return start
        return find_max(0, len(arr)-1)